edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/DeviceScopeApiClient.cs

```diff
@@ -107,7 +110,10 @@ async Task<ScopeResult> GetIdentitiesInScopeWithRetry(Uri uri)
 
         async Task<ScopeResult> GetIdentitiesInScope(Uri uri)
         {
-            var client = new HttpClient();
+            HttpClient client = null;
+            this.proxy.ForEach(p => { client = new HttpClient(new HttpClientHandler() { Proxy = p }, disposeHandler: true); });
+            client = client ?? new HttpClient();
+
```

Comment:
You should be able to `Map`/`GetOrElse` here

```suggestion
var client = this.proxy
    .Map(p => new HttpClient(new HttpClientHandler { Proxy = p }, disposeHandler: true))
    .GetOrElse(() => new HttpClient());
```